controllers/configurationpolicy_controller.go

```diff
@@ -1970,30 +2003,35 @@ func getNamesOfKind(
 	dclient dynamic.Interface,
 	complianceType string,
 	zeroValueEqualsNil bool,
-) (kindNameList []string) {
+) (kindNameList []string, allResourceList []string) {
+	var resList *unstructured.UnstructuredList
+	var err error
+
 	if namespaced {
 		res := dclient.Resource(rsrc).Namespace(ns)
 
-		resList, err := res.List(context.TODO(), metav1.ListOptions{})
-		if err != nil {
-			log.Error(err, "Could not list resources", "rsrc", rsrc, "namespaced", namespaced)
-
-			return kindNameList
-		}
+		resList, err = res.List(context.TODO(), metav1.ListOptions{})
+	} else {
+		res := dclient.Resource(rsrc)
 
-		return buildNameList(desiredObj, complianceType, resList, zeroValueEqualsNil)
+		resList, err = res.List(context.TODO(), metav1.ListOptions{})
 	}
 
-	res := dclient.Resource(rsrc)
-
-	resList, err := res.List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		log.Error(err, "Could not list resources", "rsrc", rsrc, "namespaced", namespaced)
 
-		return kindNameList
+		return kindNameList, allResourceList
+	}
+
+	if len(resList.Items) == 0 {
+		return kindNameList, allResourceList
+	}
```

Comment:
Optional: This isn't needed.
```suggestion
```

Reply:
hm.. but it goes to the unnecessary process `buildNameList`  I can remove it. It is not a big deal